src/v2/cookbook/practical-use-of-scoped-slots.md

```diff
@@ -0,0 +1,410 @@
+---
+title: Practical use of scoped slots with GoogleMaps
+type: cookbook
+order: 11
+---
+
+## Base Example
+
+There are situations when you want the template inside the slot to be able to access data from the child component that is responsible for rendering the slot content. This is particularly useful when you need freedom in creating custom templates that use the child component's data properties. That is a typical use case for scoped slots.
+
+Imagine a component that configures and prepares an external API to be used in another component, but is not tightly coupled with any specific template. Such a component could then be reused in multiple places rendering different templates but using the same base object with specific API.
+
+We'll create a component (`GoogleMapLoader.vue`) that initializes `Google Maps API`, creates a `google` and `map` objects and exposes those objects to the parent component in which the `GoogleMapLoader` is used.
+
+See below a basic example of how this can be achieved. We will analyze the code piece by piece and see what is actually happening in the next section.
+
+The template part will look as below
+
+```html
+<template>
+  <div>
+    <div id="map" :style="{ height: mapHeight }"></div>
+    <template v-if="!!this.google && !!this.map">
+      <slot
+        :google="google"
+        :map="map"
+      />
+    </template>
+  </div>
+</template>
+```
+
+Our script will look like this, we'll have some props passed to the component which allows us to set the `Google Map API` and `map` object:
+
+```js
+import GoogleMapsApiLoader from "google-maps-api-loader";
+
+export default {
+  props: {
+    mapConfig: Object,
+    apiKey: String,
+    mapHeight: String
+  },
+
+  data() {
+    return {
+      google: null,
+      map: null
+    };
+  },
+
+  async mounted() {
+    const googleMapApi = await GoogleMapsApiLoader({
+      apiKey: this.apiKey
+    });
+    this.google = googleMapApi;
+    this.initializeMap();
+  },
+
+  methods: {
+    initializeMap() {
+      const mapContainer = this.$el.querySelector("#map");
+      const { Map } = this.google.maps;
```

Comment:
I'd be careful about destructuring `Map` as this is also a native constructor in ES6. I think we shouldn't avoid shadowing these kind of objects as it might have unintended consequences and might introduce confusion. Do you agree?